frontend/src/ArticleEdit/ArticleEdit.js

```diff
@@ -52,101 +33,61 @@ class ArticleEdit extends React.Component {
     content: '',
   };
 
-  // static because - eslint-disable class-methods-use-this
-  static fetchData(id) {
-    return actions.get(id);
-  }
-
-  static getAllTopics() {
-    return getTopics();
-  }
-  static getAllSubTopics() {
-    return getSubTopics();
-  }
-
   state = {
-    edit: 0,
-    article: ArticleEdit.defaultArticle,
+    isDirty: false,
+    article: { ...ArticleEdit.defaultArticle },
     topics: [],
     sub_topics: [],
-    selectedTopic: null,
-    selectedSubTopic: ArticleEdit.nullSubTopic,
     horizontal: 'right',
     vertical: 'top',
     message: { show: false, error: false, text: '' },
   };
 
-  componentDidMount = () => {
-    const promises = [];
-    promises.push(new Promise(resolve => resolve(ArticleEdit.getAllTopics())));
-    promises.push(
-      new Promise(resolve => resolve(ArticleEdit.getAllSubTopics()))
-    );
-    if (this.props.id) {
-      promises.push(
-        new Promise(resolve => resolve(ArticleEdit.fetchData(this.props.id)))
-      );
-    }
-    Promise.all(promises).then(results => {
-      const topics = results[0];
-      const sub_topics = results[1];
-      let selectedTopic = topics[0];
-      let selectedSubTopic = ArticleEdit.nullSubTopic;
-      let article = ArticleEdit.defaultArticle;
-      if (results.length === 3) {
-        // eslint-disable-next-line prefer-destructuring
-        article = results[2];
-        selectedTopic = topics.reduce((acc, topic) => {
-          if (topic._id === article.topic._id) {
-            return topic;
-          }
-          return acc;
-        }, null);
-        if (article.sub_topic) {
-          selectedSubTopic = sub_topics.reduce((acc, sub) => {
-            if (sub._id === article.sub_topic._id) {
-              return sub;
-            }
-            return acc;
-          }, ArticleEdit.nullSubTopic);
-        } else {
-          selectedSubTopic = ArticleEdit.nullSubTopic;
-        }
+  componentDidMount() {
+    this.loadData().then(results => {
+      const [topics, sub_topics, last] = results;
+      let article;
+      if (last) {
+        article = last;
+      } else {
+        article = { ...ArticleEdit.defaultArticle, topic: topics[0] };
       }
-      this.setState({
-        article,
-        topics,
-        sub_topics,
-        selectedTopic,
-        selectedSubTopic,
-      });
+      this.setState({ topics, sub_topics, article });
     });
 
     window.addEventListener('resize', this.handleResize);
     this.handleResize();
-  };
+  }
+  // eslint-disable-next-line react/sort-comp
+  loadData() {
+    const promises = [];
+    promises.push(new Promise(resolve => resolve(getTopics())));
+    promises.push(new Promise(resolve => resolve(getSubTopics())));
+    if (this.props.id) {
+      promises.push(
+        new Promise(resolve => resolve(actions.get(this.props.id)))
+      );
+    }
+    return Promise.all(promises);
+  }
```

Comment:
Why do we need this?:
```js
new Promise(resolve => resolve(someFn()))
```
Looks like all `someFn()`s already return a promise :thinking:

Reply:
good point. will push an update.